### PR TITLE
chore(Sidebar): resolve flaky test

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-2705]

### Description

Fix flaky test in Sidebar. The problem was, that sometimes the sidebar had the collapsible button visible from the beginning as it would be hovered. So I have created another div and hover it first, to make sure, that the mouse is in the correct position.

### How to test

- tests should not fail
- expect some happo updates

### Development checks

- Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Annotate all `props` in component with documentation
- Create `examples` for component
- Ensure that deployed demo has expected results and good examples
- Ensure that tests pass by running `yarn test`
- Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
-  Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2705]: https://toptal-core.atlassian.net/browse/FX-2705?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ozgurkececioglu — APPROVED** [review]
Before the fix, my local integration tests were failing for mac and ubuntu. Both are passing now, good job 👍

File: cypress/component/PageSidebar.spec.tsx [hunk 1]
```diff
@@ -161,25 +169,28 @@ describe('Sidebar', () => {
     it('hides and shows the sidebar items text', () => {
       mount(<DefaultExample collapsible />)
 
+      // Make sure the sidebar is not hovered
+      cy.getByTestId(TestIds.CONTENT_OUTSIDE).realHover()
```
Comment (denieler) [inline cypress/component/PageSidebar.spec.tsx:173]: maybe we can just set initial mouse position to `(0, 0)`, for example, and we don't need then an additional div to be rendered

**Review by OndrejTuma — APPROVED** [review]

File: cypress/component/PageSidebar.spec.tsx [hunk 2]
```diff
@@ -161,25 +169,28 @@ describe('Sidebar', () => {
...
 
       cy.get('body').happoScreenshot({ variant: 'expanded accordion menu' })
 
       // Collapse sidebar
-      cy.get(byTestId(TestIds.SIDEBAR_CONTAINER))
+      cy.getByTestId(TestIds.SIDEBAR_CONTAINER)
         .as('container')
         .realHover()
-        .find(byTestId(TestIds.SIDEBAR_COLLAPSE_BUTTON))
+        .findByTestId(TestIds.SIDEBAR_COLLAPSE_BUTTON)
```
Comment (separatio) [inline cypress/component/PageSidebar.spec.tsx:193]: I didn't know we introduced `findByTestId`.
FYI, this is boilerplate and we don't really need it.
All test-ids should be unique, and `getByTestId` should never get 2 elements.

I suggest dropping `findByTestId` altogether and always use `getByTestId`.
It shouldn't be slower either.

File: cypress/support/commands.js [hunk 3]
```diff
@@ -20,3 +20,11 @@ Cypress.Commands.add('isWithinViewport', { prevSubject: true }, subject => {
 Cypress.Commands.add('getByTestId', (testId, options) => {
   return cy.get(`[data-testid=${testId}]`, options)
 })
+
+Cypress.Commands.add(
```
Comment (separatio) [inline cypress/support/commands.js:24]: This is boilerplate and we don't really need it.
All test-ids should be unique, and getByTestId should never get 2 elements (unless intended).

I suggest dropping findByTestId altogether and always use getByTestId.
It shouldn't be slower either.

File: cypress/component/PageSidebar.spec.tsx (hunk 1 above)
Comment (separatio) [reply cypress/component/PageSidebar.spec.tsx:173]: https://docs.cypress.io/api/commands/hover#Workarounds

Comment (separatio) [reply cypress/component/PageSidebar.spec.tsx]: As a last comment, `realHover` creates a hover event over a focused element. That means blur should work, theoretically.

File: cypress/component/PageSidebar.spec.tsx [hunk 4]
```diff
@@ -24,10 +24,10 @@ enum TestIds {
   BASIC_MENU_INNER_MENU = 'basic-menu-inner-menu',
 
   COLLAPSIBLE_MENU_HEADER = 'collapsible-header',
-  COLLAPSIBLE_MENU_INNER_MENU = 'collapsible-inner-menu'
-}
+  COLLAPSIBLE_MENU_INNER_MENU = 'collapsible-inner-menu',
 
-const byTestId = (testId: string) => `[data-testid=${testId}]`
+  CONTENT_OUTSIDE = 'content-outside'
```
Comment (separatio) [inline cypress/component/PageSidebar.spec.tsx:29]: This is not used, if I am not mistaken. Should be removed.

File: cypress/component/Select.spec.tsx [hunk 5]
```diff
@@ -339,6 +340,8 @@ describe('Select', () => {
 
     openSelect()
 
+    getOption(1).should('have.attr', 'data-highlighted').and('match', /true/)
```
Comment (separatio) [inline cypress/component/Select.spec.tsx:343]: Since we have the Happo screenshot, is this useful?

File: cypress/component/PageSidebar.spec.tsx [hunk 6]
```diff
@@ -161,25 +161,28 @@ describe('Sidebar', () => {
...
 
       cy.get('body').happoScreenshot({ variant: 'expanded accordion menu' })
 
       // Collapse sidebar
-      cy.get(byTestId(TestIds.SIDEBAR_CONTAINER))
+      cy.getByTestId(TestIds.SIDEBAR_CONTAINER)
         .as('container')
         .realHover()
-        .find(byTestId(TestIds.SIDEBAR_COLLAPSE_BUTTON))
+        .findByTestId(TestIds.SIDEBAR_COLLAPSE_BUTTON)
```
Comment (separatio) [inline cypress/component/PageSidebar.spec.tsx:185]: This should be `getByTestId`

File: cypress/component/PageSidebar.spec.tsx [hunk 7]
```diff
@@ -188,14 +191,14 @@ describe('Sidebar', () => {
...
-      cy.get(byTestId(TestIds.COLLAPSIBLE_MENU_HEADER)).realClick()
+      cy.getByTestId(TestIds.COLLAPSIBLE_MENU_HEADER).realClick()
 
       cy.get('body').happoScreenshot({ variant: 'open dropdown menu' })
 
       // Expand collapsed sidebar
       cy.get('@container')
         .realHover()
-        .find(byTestId(TestIds.SIDEBAR_COLLAPSE_BUTTON))
+        .findByTestId(TestIds.SIDEBAR_COLLAPSE_BUTTON)
```
Comment (separatio) [inline cypress/component/PageSidebar.spec.tsx:201]: This should be `getByTestId`

File: cypress/component/Select.spec.tsx (hunk 5 above)
Comment (separatio) [reply cypress/component/Select.spec.tsx:343]: Same with above assertions with `getOption`, checking if they're highlighted.

**Review by dmaklygin — APPROVED** [review]

**Review by separatio — APPROVED** [review]
Changes look good. :+1:

**Review by denieler — APPROVED** [review]

**Pull request merged**